nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/download/DownloadServiceImpl.java

```diff
@@ -233,6 +233,10 @@ public String getDownloadUrl(String storeKey) {
         if (slash < 0) {
             return null;
         }
+
+        // remove query string if any
+        path = path.replaceFirst("\\?.*$", "");
```

Comment:
Not sure there is. Why remove the query string?

Reply:
Because the download path is not expected to contain one, and that may be the case when calling `getDownloadPathAndAction` from `resolveBlobFromDownloadUrl`.